spec/inputs/snmptrap_spec.rb

```diff
@@ -1,13 +1,111 @@
 require "logstash/devutils/rspec/spec_helper"
 require "logstash/devutils/rspec/shared_examples"
+require 'logstash/plugin_mixins/ecs_compatibility_support/spec_helper'
 require 'logstash/inputs/snmptrap'
 
-describe LogStash::Inputs::Snmptrap do
+describe LogStash::Inputs::Snmptrap, :ecs_compatibility_support do
+
   it_behaves_like "an interruptible input plugin" do
     # as there is no mocking the run method will
     # raise a connection error and put the run method
     # into the sleep retry section loop
     # meaning that the stoppable sleep impl is tested
     let(:config) { {} }
   end
+
+  let(:config) { Hash.new }
+
+  subject(:input) { described_class.new(config) }
+
+  let(:source_ip) { '192.168.1.11' }
+
+  let(:snmp_manager) do
+    SNMP::Manager.new.tap do |manager|
+      def manager.send_request(trap, community, host, port)
+        trap # dummy manager - we're just using the Manager API to create traps
+      end
+    end
+  end
+
+  ecs_compatibility_matrix(:disabled, :v1, :v8) do |ecs_select|
+
+    let(:config) { super().merge 'ecs_compatibility' => ecs_compatibility }
+
+    context 'v1' do
```

Comment:
Since ECS also has `v1` and friends, clearly indicating that this `v1` refers to the SNMP version will add clarity to the specs.

```suggestion
    context 'with an SNMP v1 trap' do
```